### PR TITLE
Bump dependency versions

### PR DESCRIPTION
**Pull request by bryanlb**
Bumps several vulnerable dependencies, bumps Armeria to most recent version.

### CODE REVIEW

File: kaldb/pom.xml [hunk 1]
```diff
@@ -99,12 +99,6 @@
                 </exclusion>
             </exclusions>
         </dependency>
-        <dependency>
```
Comment (bryanlb) [inline kaldb/pom.xml:102]: We added this to the classpath to test `io_uring`, but are blocked on further unrelated infrastructure changes. Pulling out for now as this is an incubator dep, and we'll revisit once we're ready to try `io_uring` again.

File: kaldb/pom.xml [hunk 2]
```diff
@@ -15,12 +15,12 @@
         <http.client.version>4.5.13</http.client.version>
         <http.core.version>4.4.15</http.core.version>
         <protobuf.version>3.21.7</protobuf.version>
-        <grpc.version>1.47.0</grpc.version>
-        <armeria.version>1.17.0</armeria.version>
-        <micrometer.version>1.9.1</micrometer.version>
-        <kafka.version>3.2.0</kafka.version>
-        <jackson.version>2.13.3</jackson.version>
-        <jackson.databind.version>2.12.7</jackson.databind.version>
+        <grpc.version>1.49.1</grpc.version>
```
Comment (bryanlb) [inline kaldb/pom.xml:18]: Keeps grpc/micrometer consistent with what ships in Armeria.

File: kaldb/pom.xml [hunk 3]
```diff
@@ -437,6 +431,10 @@
                     <groupId>org.slf4j</groupId>
                     <artifactId>slf4j-log4j12</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>com.fasterxml.jackson.module</groupId>
+                    <artifactId>jackson-module-scala_2.13</artifactId>
```
Comment (bryanlb) [inline kaldb/pom.xml:436]: This was causing us problems attempting to target a newer databind release, and doesn't appear necessary for our use case.

File: kaldb/src/test/java/com/slack/kaldb/chunk/ReadOnlyChunkImplTest.java [hunk 4]
```diff
@@ -55,17 +55,17 @@
...
-      String.format("%sBucket", this.getClass().getSimpleName()).toLowerCase();
+  private static final String TEST_S3_BUCKET = "read-only-chunk-impl-test";
 
   private TestingServer testingServer;
   private MeterRegistry meterRegistry;
   private S3BlobFs s3BlobFs;
 
-  @ClassRule public static final S3MockRule S3_MOCK_RULE = S3MockRule.builder().silent().build();
+  @ClassRule
+  public static final S3MockRule S3_MOCK_RULE =
```
Comment (bryanlb) [inline kaldb/src/test/java/com/slack/kaldb/chunk/ReadOnlyChunkImplTest.java:67]: This is due to a change in Adobe S3Mock 2.5 that started validating against duplicate bucket names - https://github.com/adobe/S3Mock/blob/main/CHANGELOG.md#250. Since we used a class rule but tried to init the same bucket in `@Each`, this would cause subsequent tests to fail.

File: kaldb/pom.xml (hunk 2 above)
Comment (mansu) [reply kaldb/pom.xml:18]: Please make this a comment in the code so we can update it when we update armeria next.

File: kaldb/pom.xml (hunk 3 above)
Comment (mansu) [reply kaldb/pom.xml:436]: Please add this as a comment also, so we don't delete it by accident in future.

**Review by mansu — CHANGES_REQUESTED** [review]
LGTM, minor comments.

File: kaldb/pom.xml (hunk 3 above)
Comment (bryanlb) [reply kaldb/pom.xml:444]: Done.

File: kaldb/src/test/java/com/slack/kaldb/clusterManager/RecoveryTaskAssignmentServiceTest.java [hunk 5]
```diff
@@ -832,6 +833,7 @@ public void shouldHandleNodesAvailableFirstLifecycle() throws Exception {
   }
 
   @Test
+  @Ignore // Flakey, occasionally throws InternalMetadataStore on the recoveryNodeMetadataStore
```
Comment (bryanlb) [inline kaldb/src/test/java/com/slack/kaldb/clusterManager/RecoveryTaskAssignmentServiceTest.java:836]: https://github.com/slackhq/kaldb/actions/runs/3447143804/jobs/5752847656
https://github.com/slackhq/kaldb/actions/runs/3474602284/jobs/5807950536

**Review by mansu — APPROVED** [review]
